esy-package-config/Dist.re

```diff
@@ -156,7 +215,7 @@ module Parse = {
       let%bind manifest =
         switch (ManifestSpec.ofString(manifest)) {
         | Ok(manifest) => return(manifest)
-        | Error(err) => fail("invalid manifest: " ++ err)
```

Comment:
Wasn't `err` useful before?

Reply:
Previously the `git` parser was doing the parsing of urls without checking the protocols.
Now based on the protocol I am using the appropriate parser.

So previously incorrect urls like `git:git+ssh://git@github.com:22/esy/esy.git#abcdef` these were rejected by the `git` parser which used to use `gitOrGithubManifest` to parse the manifest part of the URL. (which throws the error `"missing or incorrect <manifest>"`)
Now that I am calling `gitViaSSH` parser for appropriate URL's I thought let's keep the error related to parsing manifests uniform. That's the reason I changes the error message.

you can check `gitOrGithubManifest` and `ManifestSpec.ofString` both are calling the same parsing logic internally.

So to summarise as per the test cases the `err` was not useful before, but now since I have moved around some logic this code path is being triggered and in-order to maintain uniformity in error messages I changed this line.

Let me know if this is the right thing to do.

Reply:
I'll take a closer look again, but meanwhile, could give an example?

Reply:
According to current code in master branch
```ml
let git =
    {
      let%bind () =
        ignore(string("git+ssh://"))
        <|> ignore(string("git+"))
        <|> return();
      let%bind proto =
        take_while1(c => c != ':')
        <* char(':')
        <|> fail("missing on incorrect <remote>");

      let%bind remote =
        choice(
          ~failure_msg="missing on incorrect <remote>",
          [urlWithPort, take_while1(c => c != '#' && c != ':')],
        );

      let%bind manifest = gitOrGithubManifest;
      let%bind commit = commitSHA;
      return(Git({remote: proto ++ ":" ++ remote, commit, manifest}));
    }
    <?> "<remote>(:<manifest>)?#<commit>";
```
Lets take the example of invalid uri `git:git+ssh://git@github.com:22:esy/esy.git#abcdef`
when the git parser runs, 
1. `git+ssh://` will be consumed and not bound to any identifier
2. `git@github.com` will get bound to `proto`
3. `22` will get bound to `remote`
4. when `gitOrGithubManifest` tries to parse `esy/esy.git` as manifest it fails with error

In the previous comment I was trying to say that the line in question
```ml
| Error(err) => fail("invalid manifest: " ++ err)
```
was not being used as the git parser used to handle these kinds of URLs